packages/snap-controller/src/Autocomplete/AutocompleteController.test.ts

```diff
@@ -142,6 +143,49 @@ describe('Autocomplete Controller', () => {
 		});
 	});
 
+	it('can submit with form with original query when corrected query', async () => {
+		document.body.innerHTML = '<div><form action="/search.html"><input type="text" id="search_query"></form></div>';
+		const controller = new AutocompleteController(acConfig, {
+			client: new MockClient(globals, {}),
+			store: new AutocompleteStore(acConfig, services),
+			urlManager,
+			eventManager: new EventManager(),
+			profiler: new Profiler(),
+			logger: new Logger(),
+			tracker: new Tracker(globals),
+		});
+		(controller.client as MockClient).mockData.updateConfig({ autocomplete: 'corrected', siteId: '8uyt2m' });
+
+		controller.bind();
+		let inputEl: HTMLInputElement | null;
+		let form: HTMLFormElement | null;
+		let query = 'dresss';
+
+		await waitFor(() => {
+			inputEl = document.querySelector(controller.config.selector);
+		});
+
+		form = inputEl!.form;
+
+		inputEl!.value = query;
+		inputEl!.focus();
+		inputEl!.dispatchEvent(new Event('keyup'));
+
+		await waitFor(() => {
+			controller.search();
```

Comment:
What are you waiting for here?
I think we should only be using `waitFor` with assertions.

Reply:
Pretty sure `await controller.search()` would work here.